### DISCUSSION TITLE
Components with the same name as HTML components

### DISCUSSION THREAD

**Discussion by gdcmarinho** [lit/lit#3125] [Q&A] [answered]
What is the recommendation for components with the same name as HTML components?

I'm creating a UI library and I'm able to create components with the same name as HTML components (example `<button>`), but can this get me in trouble? By the way, I didn't find any information about it in the official documentation.

**Comment by gdcmarinho** [comment]
@justinfagnani @aomarks @graynorton

**Reply by justinfagnani** [reply]
The web components spec allows for "customized built-in elements", but Safari does not and will not support that, so we don't in Lit in order to not encourage elements that would always require a polyfill in Safari.

What you should do in Lit is compose the built-in either in the shadow DOM or light DOM of your element.

**Comment by p-ob** [comment]
From MDN:

> a string representing the name you are giving to the element. Note that custom element names require a dash to be used in them (kebab-case); they can't be single words.

https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements

So I don't think this is an issue with custom element tag names.

**Reply by thescientist13** [reply]
No, they can not.  As @p-ob shared, per the specification custom element tag names must have a hyphen in the name.  All "standard" HTML element tag names are just one word.

edit: 

looks like the comment I was replying to was deleted by the OP.  here it is for posterity
> _Huh? That's not my question. I need to know if custom HTML elements can have the same name as standard HTML elements._

It was not directed at @p-ob  😅

**Reply by gdcmarinho** [reply]
Mmmmmmm... that's curious, @thescientist13. What exactly Material Design from Angular does to have a custom/standard element?

**Reply by thescientist13** [reply]
Sorry, that link doesn't work for me.  It just takes me to the stackblitz home page.

**Reply by gdcmarinho** [reply]
Uh oh! My bad, here it is:
![image](https://user-images.githubusercontent.com/34725877/178123725-ca1176f6-63ba-491f-ba94-ab7360114145.png)

**Reply by thescientist13** [reply]
Yeah, like in the case of [Angular Material Design](https://material.angular.io/components/button/overview), it's just using HTML attributes and CSS attribute selectors, in conjunction with the standard `<button>`.

> Native `<button>` and `<a>` elements are always used in order to provide the most straightforward and accessible experience for users.  A `<button>` element should be used whenever some action is performed. An `<a>` element should be used whenever the user will navigate to another view.
>
> There are several button variants, _**each applied as an attribute**_:

You could write this yourself as
```css
button[my-button] {
  background-color: red;
}
```

```html
<button my-button>Click Me</button>
```

**Comment by gdcmarinho** [comment, marked as answer]
The answer is no. Custom elements are not HTML5 default elements, what you can do is increment these standard elements with logic and stylization, but they remained standard elements.
